### PR TITLE
add set cache icon option to manage caches submenu (fix #9685)

### LINKED ISSUE #9685: Batch function to set cache icon for all caches in list
**Is your feature request related to a problem?**
I have imported a Adventure Lab GPX to see adventure lab caches around my homezone.
At the moment those are imported with the default (grey ??) icons.

In order to better identify those on the map a custom icon would be useful but needs to be set manually for all 200 caches on my list.

**Describe the solution you'd like**
The menu "Manage lists" could get an item "Set cache icons" which provides a batch function to set the icon for all caches on the list...same as already implemented for a single cache.

The reset function could however be a bit more tricky. In my case all caches have the same default icon...but that might be different for other scenarios and we should - in any case - provide some batch undo/reset in case someone uses this function accidentally.

### PR DESCRIPTION
**Pull request by moving-bits**

### CODE REVIEW

File: main/res/values/strings.xml [hunk 1]
```diff
@@ -1833,6 +1833,7 @@
         <item quantity="many">Add this cache to favorites (%d remaining)</item>
         <item quantity="other">Add this cache to favorites (%d remaining)</item>
     </plurals>
+    <string name="command_set_cache_icons_progress">Setting cache icons</string>
```
Comment (Lineflyer) [inline main/res/values/strings.xml:1836]: Maybe: "Set custom cache icons"
I would also suggest this for the cache menu: "Set custom cache icon"

Comment (Lineflyer) [reply main/res/values/strings.xml:1836]: On the other hand some translation (e.g. german=benutzerdefiniert) would then be rather long...mmh. 
Unsure..

Comment (bekuno) [reply main/res/values/strings.xml:1836]: or "Set own cache icons", or ...personal... ?

**Comment by moving-bits** [comment]
I've set the menu entry in Cache => Manage Caches to "Set cache icons" (plural) for now. We can adapt it later, when we find a good phrasing.

Also added an undo function.

**Pull request merged**